internal/fieldmask/fieldmask_test.go

```diff
@@ -0,0 +1,130 @@
+package fieldmask
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/results/internal/fieldmask/test"
+	"github.com/tidwall/gjson"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/testing/protocmp"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+)
+
+var p = []string{"a.b", "a.b.c", "d.e", "f"}
+
+var m = metadata.New(map[string]string{
+	"fields": strings.Join(p, ","),
+})
+
+var fm = FieldMask{
+	"a": FieldMask{
+		"b": {},
+	},
+	"d": FieldMask{
+		"e": {},
+	},
+	"f": {},
+}
+
+var j = `
+{ 
+	"a": { 
+		"b": { 
+			"c": "test value" 
+		} 
+	}, 
+	"d": { 
+		"e": "test value" 
+	}, 
+	"g": {
+		"h": "test value"
+	}
+}`
+
+var pm = &test.Test{
+	Id:   "test-id",
+	Name: "test-name",
+	Data: []*test.Any{
+		{
+			Type:  "type-1",
+			Value: []byte(gjson.Parse(j).String()),
+		},
+		{
+			Type:  "type-2",
+			Value: []byte(gjson.Parse(j).String()),
+		},
+	},
+}
+
+func TestNew(t *testing.T) {
+	f := &fieldmaskpb.FieldMask{Paths: p}
+	f.Normalize()
+	want := fm
```

Comment:
and any clarification you add here @sayan-biswas should become comments in the code around here, explaining expectations, http refs to field mask doc, etc.

Reply:
`a.b.(c)` is not required because the `c` is child of `b`. So if the `b` is included in the field mask then the response will include all the child and its recursive child in the response.

Reply:
thanks @sayan-biswas that helps ..... in hindsight it is obvious, just did occur to me during my last review  :-)